plugins/source/metrics.go

```diff
@@ -123,3 +124,42 @@ func (s *Metrics) TotalResourcesAtomic() uint64 {
 	}
 	return total
 }
+
+func (s *Metrics) MarkStart(table *schema.Table, clientID string) {
```

Comment:
Is this code be thread safe? It's accessed from different Go routines

Reply:
Aaaaa good point

Reply:
Added a mutex :)
We can have a different mutex per object["table"]["clientId"], because the map is never read/written concurrently (it is pre-initialized, and only the properties of the object are changed during runtime).

Thanks again for the catch, embarassing!